crates/rpc/rpc/src/eth/signer.rs

```diff
@@ -34,6 +41,24 @@ pub(crate) struct DevSigner {
     accounts: HashMap<Address, SecretKey>,
 }
 
+impl DevSigner {
+    fn sign_hash(&self, hash: H256, account: Address) -> Result<Signature> {
+        let secp = Secp256k1::new();
+        let secret =
+            self.accounts.get(&account).ok_or(RpcError::Custom("No account".to_string()))?;
```

Comment:
error can be replaced with variant.

function should have proper error response.

Reply:
Sure!
I've added a new [SignError](https://github.com/paradigmxyz/reth/pull/1665/files#diff-ae449946e8a8fa16f55c2511577cdc5a9efb699b597de667d5b39ea7542610a9R336) enum and a [new variant](https://github.com/paradigmxyz/reth/pull/1665/files#diff-ae449946e8a8fa16f55c2511577cdc5a9efb699b597de667d5b39ea7542610a9R79) for EthApiError